server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -16,183 +16,185 @@ def get(self, section: str, option: str):
 
         def mock_results_push(self, ctrl, tb, token):
-            if str(tb).startswith(str(tb_path)):
+            bad_tb = self.receive_dir / bad_tarball
+            if bad_tb == tb:
                 raise FileNotFoundError(f"No such file or directory: '{tb.name}'")
-            assert False
+            assert False, "Unexpected file or directory in mocked results_push"
 
         expected_result = Results(nstatus="", ntotal=1, ntbs=0)
         expected_error_msg = f"No such file or directory: '{bad_tarball}'"
-        monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
         monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
         assert expected_error_msg in caplog.text
 
-    def test_process_tb_connection_error(self, monkeypatch, caplog):
+    def test_process_tb_connection_error(self, monkeypatch, caplog, test_logger):
         """checks normal processing when Connection Error is faced"""
         expected_error_msg = (
             f"Cannot connect to 'https://pbench.example.com/v2/1/upload/{tarball}'"
         )
         tb_path = fixture_path / "tarball"
 
-        def mock_glob(*args, **kwargs):
-            tb_path = fixture_path / "tarball-002" / "log.tar.xz.md5"
-            return [tb_path]
-
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         def mock_results_push(self, ctrl, tb, token):
             raise RuntimeError(expected_error_msg)
 
         expected_result = Results(nstatus="", ntotal=1, ntbs=0)
-        monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
         monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
         assert expected_error_msg in caplog.text
 
-    def test_process_tb(self, monkeypatch):
+    def test_process_tb(self, monkeypatch, test_logger):
         """verify processing of tar balls without any failure"""
         tb_path = fixture_path / "tarball"
+        good_tb = tb_path / tarball
 
-        def mock_remove(*args):
-            pass
-
-        def mock_glob(*args, **kwargs):
-            tb_path = fixture_path / "tarball-002" / "log.tar.xz.md5"
-            return [tb_path]
+        def mock_remove(path, *, dir_fd=None):
```

Comment:
I think it would help me as a reviewer if you positioned the definitions of the mock functions in the order that you expect them to be invoked (so `get()` would come first and `remove()` would be last).

Reply:
Sure

Reply:
[Ditto](https://github.com/distributed-system-analysis/pbench/pull/2414#discussion_r699654394).